habitica.el

```diff
@@ -787,6 +791,31 @@ NEW-TAG is the new name to give to the tag."
   (interactive)
   (habitica-api-cron))
 
+(defun habitica-feed-pet-to-full (&optional pet food)
+  "Feed PET using FOOD until It is full."
+  (interactive)
+  (unless pet
+    (let* ((content (habitica--send-request "/content?language=en" "GET" nil))
+           (all-pets (assoc-default 'petInfo content))
+           (pet-names (mapcar #'car all-pets))
+           (pet-name (completing-read "Select the Pet: " pet-names))
+           (pet-info (cdr (assoc-string pet-name all-pets)))
+           (potion (assoc-default 'potion pet-info))
+           (all-foods (assoc-default 'food b))
+           (potion-foods (cl-remove-if-not (lambda (food-info)
+                                             (let* ((food-attrs (cdr food-info))
+                                                    (target (assoc-default 'target food-attrs)))
+                                               (string= target potion)))
+                                           all-foods))
+           (potion-food-names (mapcar #'car potion-foods))
+           (food-name (completing-read "Select the Food: " potion-food-names)))
+      (setq pet pet-name)
+      (setq food food-name))
```

Comment:
are those global variables? if so, can they be prefixed?

Reply:
no, they are arguments defined in `habitica-feed-pet-to-full (&optional pet food)`

Reply:
I'm blind apparently. Thank you!